vllm/model_executor/models/mllama.py

```diff
@@ -947,6 +947,13 @@ def forward(
             kv_cache=kv_cache,
             attn_metadata=attn_metadata,
         )
+        # the rank of full_text_row_masked_out_mask is 2, not match with
+        # the hidden_states, so expand its rank to 3.
+        # TODO: Change input_tokens tensor at the beginning of model execution
+        # to 2D tensor to align with public vllm input_tokens shape. But this
+        # will face the graph building failure issue, still need to investigate.
+        full_text_row_masked_out_mask = full_text_row_masked_out_mask.view(
+            hidden_states.size(0), -1, 1)
```

Comment:
@yisonzhu , how about only do reshape_view if the hidden_states has len(shape)==3? I think for GPU, above codes might break the model fwd?

Reply:
Done